lib/index.js

```diff
@@ -469,13 +500,33 @@ NodeHttpXHR.prototype.send = function (data) {
     this._resp = resp;
     this._responseText = '';
 
-    resp.setEncoding('utf8');
+    // var contentType = resp.headers['content-type'];
+    // TODO: adjust responseType from content-type header if applicable
+
+    // from Node API docs: The encoding argument is optional and only applies when chunk is a string. Defaults to 'utf8'.
+    if (this._responseType === 'text' || this._responseType === '') {
+      resp.setEncoding('utf8');
```

Comment:
we may need to set this for `json` case as well to actually get a string in on-data.

Reply:
or decode the buffer ourself but that seems more elegant

Reply:
Agreed - the encoding should definitely by UTF-8 for JSON.